ui/src/app/views/concept-homepage.tsx

```diff
@@ -159,17 +194,18 @@ export const ConceptHomepage = withCurrentWorkspace()(
     }
 
     componentDidMount() {
-      this.loadDomains();
+      this.loadDomainsAndSurveys();
     }
 
-    async loadDomains() {
+    async loadDomainsAndSurveys() {
       const {namespace, id} = this.props.workspace;
       try {
         const conceptsCache: ConceptCacheItem[] = [];
         const conceptDomainCounts: DomainCount[] = [];
-        const resp = await conceptsApi().getDomainInfo(namespace, id);
-        this.setState({conceptDomainList: resp.items});
-        resp.items.forEach((domain) => {
+        const [conceptDomainInfo, surveysInfo] = await Promise.all([
+          conceptsApi().getDomainInfo(namespace, id),
+          conceptsApi().getSurveyInfo(namespace, id)]);
+        conceptDomainInfo.items.map((domain) => {
```

Comment:
Sorry, I might have miscommunicated this :( 

I would have done above: `const conceptsCache: ConceptCacheItem[] = conceptDomainInfo.items.map...` and `const conceptDomainCounts: DomainCount[] = conceptDomainInfo.items.map...`